courses/management/commands/create_enrollment.py

```diff
@@ -0,0 +1,119 @@
+"""Management command to change enrollment status"""
+from django.core.management.base import BaseCommand, CommandError
+from django.contrib.auth import get_user_model
+
+from courses.models import CourseRun, CourseRunEnrollment
+from courseware.api import enroll_in_edx_course_runs
+from courseware.exceptions import (
+    EdxApiEnrollErrorException,
+    UnknownEdxApiEnrollException,
+)
+from ecommerce import mail_api
+from ecommerce.api import best_coupon_for_product
+from ecommerce.models import Coupon, Product
+from users.api import fetch_user
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """creates an enrollment for a course run"""
+
+    help = "Creates an enrollment for a course run"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--user",
+            type=str,
+            help="The id, email, or username of the User",
+            required=True,
+        )
+        parser.add_argument(
+            "--run",
+            type=str,
+            help="The 'courseware_id' value for the CourseRun",
+            required=True,
+        )
+        parser.add_argument(
+            "--code",
+            type=str,
+            help="The enrollment code for enrolling the course",
+            required=True,
+        )
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):
+        """Handle command execution"""
+
+        user = fetch_user(options["user"])
+
+        run = CourseRun.objects.filter(courseware_id=options["run"]).first()
+        if run is None:
+            raise CommandError(
+                "Could not find course run with courseware_id={}".format(options["run"])
+            )
+
+        product = Product.objects.filter(
+            courseruns__courseware_id=options["run"]
+        ).first()
+        if product is None:
+            raise CommandError(
+                "No product found for that course with courseware_id={}".format(
+                    options["run"]
+                )
+            )
+
+        try:
+            user_enrollment = CourseRunEnrollment.objects.get(user=user, run=run)
+            if user_enrollment:
+                raise CommandError(
+                    "User={} is already enrolled in courseware_id={}".format(
+                        options["user"], options["run"]
+                    )
+                )
+        except CourseRunEnrollment.DoesNotExist:
```

Comment:
Kindly, move this statement right after the `get` command in try.

Reply:
better to use the fetch and exclude the redundant `DoesNotExist` exception here.